roles/matrix-bridge-appservice-slack/defaults/main.yml

```diff
@@ -0,0 +1,92 @@
+# matrix-appservice-slack is a Matrix <-> Slack bridge
+# See: https://github.com/matrix-org/matrix-appservice-slack
+
+matrix_appservice_slack_enabled: true
+
+matrix_appservice_slack_docker_image: "cadair/matrix-appservice-slack:latest"
+
+matrix_appservice_slack_base_path: "{{ matrix_base_data_path }}/appservice-slack"
+
+# Once you make a control room in Matrix, you can get its ID by typing any message and checking its source
+slack_control_room_id: ''
+matrix_appservice_slack_bot_name: 'slackbot'
+
+# Controls the SLACK_PORT and MATRIX_PORT of the installation
+matrix_appservice_slack_matrix_port: 8008
+matrix_appservice_slack_slack_port: 9003
+
+matrix_appservice_slack_homeserver_address: 'http://matrix-synapse:{{ matrix_appservice_slack_matrix_port }}'
+matrix_appservice_slack_homeserver_domain: '{{ matrix_domain }}'
+matrix_appservice_slack_appservice_address: 'http://matrix-appservice-slack:{{ matrix_appservice_slack_matrix_port }}'
+
+# A list of extra arguments to pass to the container
+matrix_appservice_slack_container_extra_arguments: []
+
+# List of systemd services that matrix-appservice-slack.service depends on.
+matrix_appservice_slack_systemd_required_services_list: ['docker.service']
+
+# List of systemd services that matrix-appservice-slack.service wants
+matrix_appservice_slack_systemd_wanted_services_list: []
+
+matrix_appservice_slack_appservice_token: ''
+matrix_appservice_slack_homeserver_token: ''
+matrix_appservice_slack_id_token: ''
+
+matrix_appservice_slack_configuration_yaml: |
+    slack_hook_port: {{ matrix_appservice_slack_slack_port }}
+    #{{matrix_homeserver_url}} had been replaced with {{matrix_domain}} hmmmm....
+    inbound_uri_prefix: "{{ matrix_homeserver_url }}:{{ matrix_appservice_slack_slack_port }}"
```

Comment:
I'd add another variable for this:

```yaml
matrix_appservice_slack_inbound_uri_prefix: "{{ matrix_homeserver_url }}/_matrix/slack"
```

The `/_matrix/slack` thing would need to be proxied by matrix-nginx-proxy, I guess.
See the Telegram bridge's `init.yml` to see how it injects additional nginx configuration.

This would be better than exposing the whole bridge on a public URL like `https://matrix.DOMAIN:9003`.
I don't think you can even expose it like this. Even though you can expose the `9003` port, you can't make it an HTTPS listener, because the bridge doesn't have access to SSL certificates.

.. so it's best to not expose the `9003` port at all and to let matrix-nginx-proxy handle SSL and proxying.

Reply:
```
      matrix_mautrix_telegram_matrix_nginx_proxy_configuration: |
        location {{ matrix_mautrix_telegram_public_endpoint }} {
        {% if matrix_nginx_proxy_enabled|default(False) %}
        	{# Use the embedded DNS resolver in Docker containers to discover the service #}
        	resolver 127.0.0.11 valid=5s;
        	set $backend "matrix-mautrix-telegram:8080";
        	proxy_pass http://$backend;
        {% else %}
        	{# Generic configuration for use outside of our container setup #}
        	proxy_pass http://127.0.0.1:9006;
        {% endif %}
        }
```

I'm new to all of this, so bear with me
What should LOCATION `(matrix_mautrix_telegram_public_endpoint)` be for an appservice? Does matrix-appservice-slack need a specific location?

Reply:
I think you just need to match whatever you pass as the path in `inbound_uri_prefix`.

Example:

```yaml
matrix_appservice_slack_public_endpoint: /appservice-slack

matrix_appservice_slack_inbound_uri_prefix: "{{ matrix_homeserver_url }}{{ matrix_appservice_slack_public_endpoint }}"
```

`config.yaml`:

```yaml
inbound_uri_prefix: {{ matrix_appservice_slack_inbound_uri_prefix }}
```

And then for nginx you can use `location {{ matrix_appservice_slack_public_endpoint }}`

Reply:
So the endpoint **won't** be /_matrix/slack, but instead /appservice-slack? I assume it doesn't matter what the endpoint is as long as it matches the nginx proxy?

Reply:
That's right. It doesn't matter what it is, as long as it's the same.

I first thought about `/_matrix/slack`, but I guess `/appservice-slack` is better. Putting it under `/_matrix` makes it sound like it's part of the Matrix Client-Server API or something of that sort. Well, that one is at `/_matrix/client`, so there is no confusion, but still..

Maybe in the future we'll support another Slack bridge (different than AppService Slack) and this one using something like `/slack` seems wrong. Better to use the full name (`appservice-slack`) anyway, regardless if it's under `/_matrix` or not.

Reply:
I tested out the current version today and I got an error from matrix-synapse.confd. It included the block:
```
[
a
n
s
i
b
l
e
u
n
d
e
f
i
n
e
d
]
```

Is this a fault of my nginx setup or something else?

Reply:
The block was part of the configuration or you got it as an error message when you ran the playbook?

Reply:
It was part of the configuration, specifically `matrix-synapse.confd`

Reply:
Looks like it's because `matrix_mautrix_telegram_matrix_nginx_proxy_configuration` needs to be changed to `matrix_appservice_slack_matrix_nginx_proxy_configuration`